### PR TITLE
MUL-1054 Exchangelly 3rd-party integration

### PR DESCRIPTION
**Pull request by nickpro**
- Integrated Exchangelly 3rd party
 - Added N exchangers configuration possibility
 - Unit test and integration test coverage
 - Func test in python repo added
 - Added rest api handlers with strict changelly usage

### CODE REVIEW

File: client/rest-exchangers.go [hunk 1]
```diff
@@ -0,0 +1,121 @@
+/*
+Currency implementation is abstractly support multiple exchangers functionality, so API user can compare
+available exchanger services and choose any.
+For not we support only Changelly exchanger 3rd party.
+ */
```
Comment (jekabolt) [inline client/rest-exchangers.go]: at the top of any code file we need a short header about license

File: client/rest-exchangers.go [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+package client
```
Comment (jekabolt) [inline client/rest-exchangers.go]: same

File: exchanger/factory_test.go [hunk 3]
```diff
@@ -0,0 +1,40 @@
+package exchanger
```
Comment (jekabolt) [inline exchanger/factory_test.go]: licence header

File: client/rest-exchangers.go (hunk 2 above)
Comment (nickpro) [reply client/rest-exchangers.go]: sure, let me fix

File: client/rest-exchangers.go (hunk 1 above)
Comment (nickpro) [reply client/rest-exchangers.go]: done

File: client/rest-exchangers.go [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+
+		var requestData RequestGetExchangeAmount
+		err := decodeBody(c, &requestData)
+		if err != nil {
+			restClient.log.Errorf("GetExchangerAmountExchange: Failed to decode body: %s\t[addr=%s]",
+				err.Error(), c.Request.RemoteAddr)
+			c.JSON(http.StatusBadRequest, gin.H{"code": http.StatusBadRequest, "message": msgErrRequestBodyError})
+			return
+		}
+		changellyExchanger, _ := restClient.
```
[the remark this answers is not in the document]
Comment (nickpro) [reply client/rest-exchangers.go]: coz for now we do not support multiply exchangers, so we create 1 with fixed name and then get it from factory with the same fixed name

File: exchanger/factory_test.go (hunk 3 above)
Comment (nickpro) [reply exchanger/factory_test.go]: done

File: multy-back.go [hunk 5]
```diff
@@ -102,9 +106,33 @@ func Init(conf *Configuration) (*Multy, error) {
...
+
 	return multy, nil
 }
 
+func (m *Multy) ConfigureExchangers() error {
+	m.ExchangerFactory.SetExchangersConfig(m.config.Exchangers)
+
+	for _, exchangerConfig := range m.config.Exchangers {
+		if exchangerConfig.IsActive {
+			// exchanger warm-up
```
[the remark this answers is not in the document]
Comment (nickpro) [reply multy-back.go]: ?

File: client/rest-exchangers.go [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+	}
+}
+
+func (restClient *RestClient) CreateExchangerTransaction() gin.HandlerFunc {
+	return func(c *gin.Context) {
+		type RequestCreateTransaction struct {
+			From string			`json:"from"`
+			To string			`json:"to"`
+			Amount float64		`json:"amount"`
+			Address string		`json:"address"`
```
Comment (jekabolt) [inline client/rest-exchangers.go]: go fmt

File: exchanger/common.go [hunk 7]
```diff
@@ -0,0 +1,29 @@
+/*
```
Comment (jekabolt) [inline exchanger/common.go]: common.go

File: client/rest-exchangers.go (hunk 6 above)
Comment (nickpro) [reply client/rest-exchangers.go]: done

File: client/rest.go [hunk 8]
```diff
@@ -145,6 +150,10 @@ func SetRestHandlers(
 		v1.GET("/exchange/changelly/list", restClient.changellyListCurrencies())
```
Comment (jekabolt) [inline client/rest.go]: we can now remove this

File: exchanger/changelly_api.go [hunk 9]
```diff
@@ -0,0 +1,210 @@
...
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"strconv"
+)
+
+const (
+	ExchangeChangellyCanonicalName  = "changelly"
+	RpcGetCurrencies = "getCurrencies"
```
Comment (jekabolt) [inline exchanger/changelly_api.go]: why rpc?

File: exchanger/changelly_api.go [hunk 10]
```diff
@@ -0,0 +1,210 @@
...
+type InitConfig struct {
+	ApiUrl string
+	ApiKey string
+	ApiSecret string
+}
+
+type rpcPacket struct {
+	Id string					`json:"id"`
+	Jsonrpc string				`json:"jsonrpc"`
+	Method string				`json:"method""`
```
Comment (jekabolt) [inline exchanger/changelly_api.go]: go fmt ./...

File: client/rest.go (hunk 8 above)
Comment (nickpro) [reply client/rest.go]: done

**Pull request merged**